update Uptycs jwt token authentication and add new commands

cgadde-uptycs opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
1) Updated the jwt token authentication module as per latest Uptycs public api
2) Added new commands to enhance user experience

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=27003) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=27003) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=27003) it.</sub>

RotemAmit commented:
> Thank you for your contribution. Your generosity and caring are unrivaled! Make sure to register your contribution by filling the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form, so our content wizard @RotemAmit will know the proposed changes are ready to be reviewed.

Thank you for your contribution! Please register your contribution by filling the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form. Thank you!

RotemAmit commented:
> > Thank you for your contribution. Your generosity and caring are unrivaled! Make sure to register your contribution by filling the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form, so our content wizard @RotemAmit will know the proposed changes are ready to be reviewed.
> 
> Thank you for your contribution! Please register your contribution by filling the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form. Thank you!

Thank you for your contribution!
This is a kindly reminder to register your contribution by filling the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form. Thank you again.

RotemAmit commented on Packs/Uptycs/Integrations/Uptycs/README.md:
```diff
@@ -2578,6 +2592,698 @@ There is no context output for this command.
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| table_id | id of the table | Required |
```
```suggestion
| table_id | table id | Required |
```

RotemAmit commented on Packs/Uptycs/Integrations/Uptycs/README.md:
```diff
@@ -2578,6 +2592,698 @@ There is no context output for this command.
+|TRUE|2023-04-21T08:27:20.888Z|f976bda8-d5dc-468f-8283-20d5368352e2|b1c3b08c-eedd-4b94-8ba0-9ca322401016|null|null|SELECT id_field_value,data FROM upt_lookup_rows WHERE lookup_table_id = '3fdb051b-b38b-4792-9daa-0a88ba4fae53'|TRUE|3fdb051b-b38b-4792-9daa-0a88ba4fae53|remote_address|Test 1|24|null|2023-04-25T04:11:04.664Z|f976bda8-d5dc-468f-8283-20d5368352e2|
+
+
+### 44. uptycs-get-lookuptables
```
```suggestion
### 45. uptycs-get-lookuptables
```

RotemAmit commented on Packs/Uptycs/Integrations/Uptycs/Uptycs.yml:
```diff
@@ -1798,7 +1798,539 @@ script:
+      default: false
+    - name: tag_id
+      required: true
+      description: tag id
```
```suggestion
      description: Disassociate the tag matching this id with the given asset_id
```

RotemAmit commented on Packs/Uptycs/Integrations/Uptycs/README.md:
```diff
@@ -2578,6 +2592,698 @@ There is no context output for this command.
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| tag_id | disassociate the tag matching this id | Required |
+| asset_id | disassociate the asset matching this asset id | Required |
```
```suggestion
| tag_id | Disassociate the tag matching this id with the given asset_id | Required |
| asset_id | Disassociate the asset matching this asset id with the given tag_id | Required |
```

RotemAmit commented on Packs/Uptycs/Integrations/Uptycs/Uptycs.yml:
```diff
@@ -1798,7 +1798,539 @@ script:
+    arguments:
+    - name: asset_id
+      required: true
+      description: asset id
```
```suggestion
      description: Disassociate the asset matching this asset id with the given tag_id
```

RotemAmit reviewed (commented):
Thank you for your contribution! Nice work!
A few comments:
* Please make sure that the description of the commands and arguments in the YML matches the README.
* Please add unit tests for the new commands. You can use the following guide - https://xsoar.pan.dev/docs/integrations/unit-testing as well as examples of unit tests in the HelloWorld integration and other packs from the content repository.

RotemAmit reviewed (commented):
Thank you for your contribution! Nice work!
A few comments:
* Please make sure that the description of the commands and arguments in the YML matches the README.
* Please add unit tests for the new commands. You can use the following guide - https://xsoar.pan.dev/docs/integrations/unit-testing as well as examples of unit tests in the HelloWorld integration and other packs from the content repository.

cgadde-uptycs commented:
@RotemAmit updated all description as per your comments. Thank you.

RotemAmit reviewed (commented):
Nice work!
* Please add unit tests for the new commands. You can use the following guide - https://xsoar.pan.dev/docs/integrations/unit-testing as well as examples of unit tests in the HelloWorld integration and other packs from the content repository.
* Please make sure that the description of the Context Output in the README and YML files matches.
If you have any questions feel free to write me here or via DFIR.

cgadde-uptycs commented:
@RotemAmit  I synced all description in yml and Readme. Added a testcase, Can you please check and let me know if this is fine ?  If this is fine, I can add remaining command testcases.

RotemAmit commented:
> @RotemAmit I synced all description in yml and Readme. Added a testcase, Can you please check and let me know if this is fine ? If this is fine, I can add remaining command testcases.

Hi, overall the test looks ok, but it fails when running pytest. You can find the error [here](https://app.circleci.com/pipelines/github/demisto/content/272686/workflows/4db714ce-bdbc-4193-91c6-72c27cae4b05/jobs/706410), please take a look and fix it. Moreover, there is a Flake8 error that you can see in the same link. Please fix it too.

cgadde-uptycs commented:
@RotemAmit Need you help. mocker used in tests is unable to set mandatory parameters domain and customer_id, Hence testcases are failing. 2 other parameters key and secret were getting set.

RotemAmit commented:
> @RotemAmit Need you help. mocker used in tests is unable to set mandatory parameters domain and customer_id, Hence testcases are failing. 2 other parameters key and secret were getting set.

You can try to mock all the global variables inside the tests as well, like in the following example: 
mocker.patch("Uptycs.CUSTOMER_ID", new=CUSTOMER_ID_VALUE)

cgadde-uptycs commented:
@RotemAmit Completed adding testcases. Please check.

RotemAmit commented on Packs/Uptycs/Integrations/Uptycs/Uptycs.py:
```diff
@@ -2199,17 +2249,131 @@ def uptycs_post_saved_query_command():
+        if len(query_results.get('items')) == 0:
+            return False
         return True
+    else:
+        return False
```
If a command fail (in this case the test_module), it should raise an exception with an informative message with the failing reason. Please update this.

RotemAmit commented on Packs/Uptycs/ReleaseNotes/1_0_9.md:
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Uptycs
+- updated Uptycs jwt token authentication and added new commands.
```
Please add a list of the new commands, as in the example in the screenshot.
![image](https://github.com/demisto/content/assets/71597826/119d7c38-396c-4f6d-ad91-51db24f290b3)

cgadde-uptycs replied:
updated

RotemAmit commented on Packs/Uptycs/ReleaseNotes/1_0_9.md:
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Uptycs
+- updated Uptycs jwt token authentication and added new commands.
```
```suggestion
- updated Uptycs jwt token authentication and added new commands.
- Updated the Docker image to: *demisto/uptycs:1.0.0.63766*.
```
Please also add a line on the updated docker image.

cgadde-uptycs replied:
updated

RotemAmit reviewed (commented):
Nice work!
* Please add unit tests to the commands uptycs-post-lookuptable-data and to the command uptycs-create-lookuptable .
* Let's schedule a demo in order make sure that everything works as expected. I am sending you an email.

cgadde-uptycs commented:
@RotemAmit Main Uptycs library was unable to pick file from test_data, Can you please tell me how to handle it ?
![image](https://github.com/demisto/content/assets/106585855/c69ffd8a-2c30-4a37-8345-b2ddc02a104f)

RotemAmit commented:
> @RotemAmit Main Uptycs library was unable to pick file from test_data, Can you please tell me how to handle it ? ![image](https://user-images.githubusercontent.com/106585855/249688591-c69ffd8a-2c30-4a37-8345-b2ddc02a104f.png)

Hi,
You need to mock demisto.getFilePath, just like you mocked demisto.args. It's return value should be like this: {'id': id, 'path': 'test/test.txt', 'name': 'test.txt'}.

cgadde-uptycs commented:
@RotemAmit Thank you for the help. Completed adding testcase for the remaining 2 commands.

cgadde-uptycs commented:
@RotemAmit As discussed during the demo. I made the following changes.

1) For command "uptycs-edit-lookuptable" , argument "active" was not working. Fixed it :
![Screenshot from 2023-07-03 17-32-15](https://github.com/demisto/content/assets/106585855/d7815797-4003-4244-9cab-d05e153bd52a)

2) Added error message for 403 response for command 'uptycs-get-carves-download-file' :
![Screenshot from 2023-07-03 17-29-11](https://github.com/demisto/content/assets/106585855/a4d7c8fd-6038-4c10-b1a7-33728c17ded8)

3) For command uptycs-get-carves , optional argument path was not working. Fixed it : 
![Screenshot from 2023-07-03 17-25-36](https://github.com/demisto/content/assets/106585855/4d89c8fb-eb8b-4609-b399-879e56aae1e0)

RotemAmit reviewed (commented):
Nice work!
Just a few last comments.

cgadde-uptycs commented on Packs/Uptycs/Integrations/Uptycs/Uptycs.yml:
```diff
@@ -1798,7 +1798,544 @@ script:
+    arguments:
+    - name: asset_id
+      required: true
+      description: Uptycs asset id
```
description updated.

cgadde-uptycs commented on Packs/Uptycs/Integrations/Uptycs/Uptycs.yml:
```diff
@@ -1798,7 +1798,544 @@ script:
+    arguments:
+    - name: tag_id
+      required: true
+      description: Uptycs tag id
```
description updated.

cgadde-uptycs commented on Packs/Uptycs/Integrations/Uptycs/Uptycs.yml:
```diff
@@ -1798,7 +1798,544 @@ script:
+      default: false
+    - name: id_field
+      required: true
+      description: Id field for the table
```
description updated.

cgadde-uptycs commented on Packs/Uptycs/Integrations/Uptycs/Uptycs.yml:
```diff
@@ -1798,7 +1798,544 @@ script:
+    arguments:
+    - name: table_id
+      required: true
+      description: Id of table in Uptycs DB
```
description updated.

cgadde-uptycs commented:
@RotemAmit descriptions updated as suggested. Thanks for the inputs.

RotemAmit commented on Packs/Uptycs/Integrations/Uptycs/README.md:
```diff
@@ -2578,6 +2592,726 @@ There is no context output for this command.
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| tag_id | Uptycs tag id. Get id from list of tags with command uptycs-get-tags. | Required |
```
```suggestion
| tag_id | Uptycs tag id. Get the id from the list of tags with the command uptycs-get-tags. | Required |
```
Please update this change in all of the relevant places in the yml and README.

RotemAmit commented on Packs/Uptycs/Integrations/Uptycs/README.md:
```diff
@@ -2578,6 +2592,726 @@ There is no context output for this command.
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| table_id | Look up table id. Get id from list of tables with command uptycs-get-lookuptables.| Required |
```
```suggestion
| table_id | Look up table id. Get the id from the list of tables with the command uptycs-get-lookuptables.| Required |
```
Please update this change in all of the relevant places in the yml and README.

RotemAmit reviewed (commented):
Nice work!

cgadde-uptycs commented:
@RotemAmit updated descriptions as per comments. Thank you.
